home_assistant_streamdeck_yaml.py

```diff
@@ -1536,7 +1982,118 @@ async def _sync_input_boolean(
             {},
             {"entity_id": state_entity_id},
         )
+        
+def _on_touchscreen_event_callback(
+    websocket: websockets.WebSocketClientProtocol,
+    complete_state: StateDict,
+    config: Config,
+):
+    async def touchscreen_event_callback(
+        deck: StreamDeck,
+        event_type: TouchscreenEventType,
+        value: dict[str, int]
+    ):
+        console.log(f"Touchscreen event {event_type} called at value {value}")
+        if event_type == TouchscreenEventType.DRAG:
+            # go to next or previous page
+            if value["x"] > value["x_out"]:
+                console.log(f"Going to page {config.next_page_index}")
+                config.to_page(config.next_page_index)
+                
+            else:
+                console.log(f"Going to page {config.next_page_index}")
+                config.to_page(config.previous_page_index)
+            deck.reset()
+            config.current_page().sort_dials()
+            update_all_key_images(deck, config, complete_state)
+            update_all_dials(deck, config, complete_state)
+        else:
+            # sets value of dial to maximum value
+            lcd_icon_size = deck.touchscreen_image_format()["size"][0] / deck.dial_count()
+            icon_pos = value["x"] // lcd_icon_size
+            dials = config.dial_sorted(int(icon_pos))
+
+            if (dials[0].dial_event_type == DialEventType.TURN.name):
+                selected_dial = dials[0]
+            else:
+                selected_dial = dials[1]
+            if selected_dial.allow_touchscreen_events:
+                if event_type == TouchscreenEventType.SHORT:
```

Comment:
Remove duplicated code, only get min or max in the condition and call `set_state` and `handle_dial_event` afterwars (outside the condition)